### DISCUSSION TITLE
center div flexbox

### DISCUSSION THREAD

**Discussion by maxdnc** [cssbattle/forum#313] [General]
Hello ! I'm new here :)
I'm trying to center a `<div>` in the middle of the screen ( vertical and horizontal ) but I get only the horizontal center. why ?
 What am i doing wrong ? thanks for your help 👍 
`
```
<div id="shape">
  <div class="square1"></div>
  <div class="square2"></div>
  <div class="square3"></div>
</div>

<style>
  body {
    background: #62306D;
  }
  
  #shape {
    background: black; 
    display: flex;
    justify-content: center;
    align-items: center;
  }
  
.square1{
  width: 50px;
  height: 50px;
  background: #F7EC7D;
}
  .square2{
  width: 50px;
  height: 50px;
  background: #F7EC7D;
}
  .square3{
  width: 50px;
  height: 50px;
  background: #F7EC7D;
}
```

**Comment by Beowolve** [comment]
`align-items:center` does perfectly align your squares centered inside #shape. You see that if you add `height:100%` to #shape. By default a div has no height. If you instead want the #shape container centered you would need to rewrite it, like so for example:

```
  body {
    background: #62306D;
    display: flex;
    align-items: center;
  }
 
  #shape {
    background: black; 
    display: flex;
    justify-content: center;
    width:100%;
  }
  ```

**Reply by maxdnc** [reply]
thank you ! 
ahhhh I get it ! the div has no height ! that why it didn't align in the center of the screen at first, so I need to do it from the body !
thankkk 👯‍♂️
